source/change-streams.rst

```diff
@@ -339,19 +435,44 @@ Single Server Topologies
 
 Presently, change streams cannot be initiated on single server topologies as they do not have an oplog.  Drivers MUST NOT throw an exception in this scenario, but instead rely on an error returned from the server.  This allows for the server to seamlessly introduce support for this in the future, without need to make changes in driver code.
 
+startAtClusterTime
+^^^^^^^^^^^^^^^^^^
+
+:since: 4.0
+
+``startAtClusterTime`` specifies that a ``changeStream`` will only return changes that occurred after the specified timestamp.
+
+If the option ``startAtClusterTime`` is not specified, and the server version is >= ``4.0`` drivers MUST set a default ``startAtClusterTime`` value using a ``$clusterTime`` from any server response (ex: ``ismaster``).
+
+``startAtClusterTime`` and ``resumeAfter`` are mutually exclusive; if both ``startAtClusterTime`` and ``resumeAfter`` are set, the server will return an error. Drivers MUST NOT throw a custom error, and MUST defer to the server error.
+
 resumeAfter
 ^^^^^^^^^^^
 
-When resuming a change stream after a disconnect, the driver issuing a new ``$changeStream`` request MUST specify a ``resumeAfter`` key with a resume token from the last change it saw.  In this case, the aggregation will return notifications starting with the log entry immediately *after* the provided token.  If the resume token specified does not exist, the server will return an error.  If ``resumeAfter`` is omitted completely, or is null, the most recent oplog entry will be returned.
+:since: 3.6
+
+``resumeAfter`` is used to resume a changeStream that has been stopped to ensure that only changes starting with the log entry immediately *after* the provided token will be returned. If the resume token specified does not exist, the server will return an error. 
 
 Resume Process
 ^^^^^^^^^^^^^^
 
-Once a ``ChangeStream`` has encountered a resumable error, it MUST attempt to resume one time.  The process for resuming MUST follow these steps:
+Once a ``ChangeStream`` has encountered a resumable error, it MUST attempt to resume one time. The process for resuming MUST follow these steps:
+
+- Perform server selection.
+- Connect to selected server.
+- If the ``ChangeStream`` has not received any changes, and the server version is >= ``4.0``:
 
-- Perform server selection
-- Connect to selected server
-- Execute the known aggregation command, specifying a ``resumeAfter`` with the last known ``resumeToken``
+    - The driver MUST execute the known aggregation command.
```

Comment:
I realize this wasn't in the original text, but would it be helpful to mention that the `aggregate` command for the resume attempt MUST use the same session (be it explicit or implicit) as the previous attempt's command?

Reply:
done